git_machete/tests/functional/test_machete.py

```diff
@@ -1645,6 +1644,7 @@ def test_anno_prs(self) -> None:
     git_api_state_for_test_create_pr = MockGithubAPIState([{'head': {'ref': 'ignore-trailing'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'hotfix/add-trigger'}, 'number': '3', 'html_url': 'www.github.com'}])
 
     @mock.patch('git_machete.options.CommandLineOptions', FakeCommandLineOptions)
+    @mock.patch('git_machete.github.GITHUB_REMOTE_PATTERNS', FAKE_GITHUB_REMOTE_PATTERNS)
```

Comment:
Hmmm why is mocking `GITHUB_REMOTE_PATTERNS` necessary? 🤔

Reply:
This is to bypass error: `Remotes are defined for this repository, but none of them corresponds to GitHub`. I could simply add another remote which corresponds to Github but then in `test_checkout_pr` i would never reach correct repository name. To sum up, this is only to show up correct repository name in error message while testing wrong pull request number.

Reply:
(I'd intuitively still favor adding an actual github remote to reduce the amount of mocking... but maybe that's indeed gonna end up more complicated)

Reply:
yes, this will work on other github methods tests but not in checkout-pr test. Even if I grab exact repository name of `tmp.XXXXXX`, and add new remote with url `https://github.com/user/repo_name.git` it will fail on `fetch` because given url does not exists. `FAKE_GITHUB_REMOTE_PATTERNS` allows to select local (non hosted) repository for fetching.

Reply:
Awwww okay I see, it's the **fetch** that's the problem specific to `checkout-pr(s)`... okay!